### ISSUE TITLE
Official Docker container

### ISSUE THREAD

**Issue by jcpunk** [tinyproxy/tinyproxy#501]
## Tinyproxy version
1.11.1

## Issue
With the lightweight footprint of tinyproxy, it would be a helpful tool to deploy when troubleshooting network connectivity issues.  Having an official container would simplify deployment and ensure folks are using known approved builds.

This project may be a candidate for the docker library of official images.

**Comment by denis-domanskii** [comment]
I fully understand philosophy of authors of this library - you spend hundreds of hours to make it really tiny but functional. And Docker image seems (from your perspective) as a fat cross on the main idea of your work.

But I run it in Docker now. Because main value of you app for me is not its size, but easy configuration and stable in work. And I need Docker to maintain my home lab infrastructure in a unify way, where all apps are containers, with config, control and backups in one place (Portainer). Also I have 2 dedicated servers, and it's much easier just to press "Deploy app on server B" app in UI then connect via SSH and deploy/config/run it manually. 

So, there is no right answer here. You doing this software for fun, so it's fully up to you what you want or don't want to support in it. From the other hand, it's also important how people use you product. And if 90% of users use your microscope as a hummer - then maybe you need to attach the handle :)

P.S. Now I use [kalaksi/docker-tinyproxy ](https://github.com/kalaksi/docker-tinyproxy) as container and it works well.

**Comment by DeadNews** [comment]
>the lightweightness is gone when you run it through docker - instead of a 100 kb binary you now gotta load an entire OS.

Not the entire OS, only the necessary dependencies.

You can use [distroless images](https://github.com/GoogleContainerTools/distroless).

Or make your own [from scratch](https://docs.docker.com/build/building/base-images/#create-a-base-image).

**Comment by DeadNews** [comment]
>the lightweightness is gone when you run it through docker - instead of a 100 kb binary you now gotta load an entire OS.

Not the entire OS, only the necessary dependencies.

You can use [distroless images](https://github.com/GoogleContainerTools/distroless).
Or make your own [from scratch](https://docs.docker.com/build/building/base-images/#create-a-base-image).

**Comment by DeadNews** [comment]
>the lightweightness is gone when you run it through docker - instead of a 100 kb binary you now gotta load an entire OS.

Not an entire OS, only the necessary runtime dependencies.

You can use [distroless images](https://github.com/GoogleContainerTools/distroless).
Or make your own [from scratch](https://docs.docker.com/build/building/base-images/#create-a-base-image).

**Comment by obnoxxx** [comment]
I agree that it could be convenient to have an "official"container image for tinyproxy.


In the spirit of true open sorce, I would not advocate  for docker though but maybe for an image to be used with podman. In particular, I would not ask docker folks to create such an image but would suggest to have the tinyproxy developer community to provide the Containerfile and possibly host images on quay.io. 

As I have some experience  with this kind of stuff from other projects, I will be happy to spend some cycles on creating an initial draft for a container image inside the code repo.

An alternative to building directly from sources in the container is to consume distro packages.  Several popular linux distributions have high quality tinyproxy packages that could be installed into the container directly.

@rofl0r - wdyt? - would you be opposed to the approach?

**Comment by rofl0r** [comment]
@obnoxxx i'm not familiar with podman so i can't comment on the technicalities, but if you think it's a good idea i'm not opposed to it - as long as you or someone else skilled in that area keeps maintaining it. though it does sound like something that would be better kept in a separate repository inside the organization in order to prevent the git history from becoming cluttered with changes to those build scripts or whatever is required to describe the container. as for whether to use distro packages or master is a two-edged sword - distro packages don't provide bugfixes apart from security relevant ones, but they're safe from commits that introduce regressions. otoh *if* there are regressions in master, it could be advantageous (although inconvenient for them) if users run into them and can provide early feedback.

**Comment by schildbach** [comment]
For what it's worth, post my personal Dockerfile here. It's customized to my usecase (most featured disabled), and it's distroless (just contains/runs the tinyproxy binary and that's it). I'm pretty confident it could be built and run with Podman as well, or buildah, buildx.

```
FROM debian:bookworm-slim AS build-stage

ENV DEBIAN_FRONTEND noninteractive
RUN apt-get update && \
    apt-get --yes upgrade && \
    apt-get --yes install git build-essential automake && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*

RUN adduser --shell /bin/false builder
USER builder

WORKDIR /home/builder
ARG GIT_REPO=https://github.com/tinyproxy/tinyproxy.git
ARG GIT_TAG=1.11.2
RUN git clone --depth 1 $GIT_REPO --branch $GIT_TAG tinyproxy

WORKDIR /home/builder/tinyproxy
RUN ./autogen.sh && \
    LDFLAGS=-static ./configure \
      --prefix= \
      --disable-xtinyproxy \
      --disable-filter \
      --disable-upstream \
      --disable-reverse \
      --disable-transparent \
      --disable-manpage_support
RUN make

FROM scratch AS run-stage

ARG USER_ID=1000

COPY --from=build-stage --chown=0 /home/builder/tinyproxy/src/tinyproxy /
COPY --chown=${USER_ID} default-tinyproxy.conf /tinyproxy.conf

USER ${USER_ID}
CMD ["/tinyproxy", "-d", "-c", "/tinyproxy.conf"]
```
